hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeDecommissionMetrics.java

```diff
@@ -0,0 +1,245 @@
+        .set(numTrackedUnhealthy);
+  }
+
+  public void setTrackedContainersSufficientlyReplicatedTotal(
+          long numTrackedSufficientlyReplicated) {
+    trackedContainersSufficientlyReplicatedTotal
+        .set(numTrackedSufficientlyReplicated);
+  }
+
+  public long getTrackedDecommissioningMaintenanceNodesTotal() {
+    return trackedDecommissioningMaintenanceNodesTotal.value();
+  }
+
+  public long getTrackedRecommissionNodesTotal() {
+    return trackedRecommissionNodesTotal.value();
+  }
+
+  public long getTrackedPipelinesWaitingToCloseTotal() {
+    return trackedPipelinesWaitingToCloseTotal.value();
+  }
+
+  public long getTrackedContainersUnderReplicatedTotal() {
+    return trackedContainersUnderReplicatedTotal.value();
+  }
+
+  public long getTrackedContainersUnhealthyTotal() {
+    return trackedContainersUnhealthyTotal.value();
+  }
+
+  public long getTrackedContainersSufficientlyReplicatedTotal() {
+    return trackedContainersSufficientlyReplicatedTotal.value();
+  }
+
+  public void metricRecordPipelineWaitingToCloseByHost(String host,
+                                                       long num) {
+    trackedPipelinesWaitingToCloseByHost.computeIfAbsent(host,
+        hostID -> registry.newGauge(
+            Interns.info("trackedPipelinesWaitingToClose-" + hostID,
+                "Number of pipelines waiting to close for "
+                    + "host in decommissioning and "
+                    + "maintenance mode"), 0L)).set(num);
+  }
+
+  public void metricRecordOfReplicationByHost(String host,
+                                              long sufficientlyReplicated,
+                                              long underReplicated,
+                                              long unhealthy) {
+    trackedSufficientlyReplicatedByHost.computeIfAbsent(host,
+        hostID -> registry.newGauge(
+            Interns.info("trackedSufficientlyReplicated-" + hostID,
+                "Number of sufficiently replicated containers "
+                    + "for host in decommissioning and "
+                    + "maintenance mode"), 0L)).set(sufficientlyReplicated);
+
+    trackedUnderReplicatedByHost.computeIfAbsent(host,
+        hostID -> registry.newGauge(
+            Interns.info("trackedUnderReplicated-" + hostID,
+                "Number of under-replicated containers "
+                    + "for host in decommissioning and "
+                    + "maintenance mode"), 0L)).set(underReplicated);
+
+    trackedUnhealthyContainersByHost.computeIfAbsent(host,
+        hostID -> registry.newGauge(
+            Interns.info("trackedUnhealthyContainers-" + hostID,
+                "Number of unhealthy containers "
+                    + "for host in decommissioning and "
+                    + "maintenance mode"), 0L)).set(unhealthy);
+  }
+
+  @VisibleForTesting
+  public Long getTrackedPipelinesWaitingToCloseByHost(String host) {
+    if (!trackedPipelinesWaitingToCloseByHost.containsKey(host)) {
```

Comment:
These get methods could all be simplified to something like:

```
private static final MutableGaugeLong ZERO_GAUGE = new MutableGaugeLong;

return trackedPipelinesWaitingToCloseByHost.getOrDefault(host, ZERO_GAUGE).value()
```

Reply:
getters from metrics by host cleaned up in latest commit.  `getTrackedPipelinesWaitingToCloseByHost` , please check.